{
  "smartbugs_completed": true,
  "analyzer_results": {
    "slither": {
      "successfull_analysis": true,
      "errors": [
        "EXIT_CODE_6"
      ],
      "vulnerability_findings": [
        {
          "name": "external-function",
          "vulnerability_from_line": 6,
          "vulnerability_to_line": null,
          "vulnerability_code": "    function() public payable {}",
          "message": "MultiplicatorX4.fallback (patch_5.sol#6) should be declared external\n"
        },
        {
          "name": "external-function",
          "vulnerability_from_line": 8,
          "vulnerability_to_line": 11,
          "vulnerability_code": "    function withdraw() public payable {\n        require(msg.sender == owner);\n        owner.transfer(this.balance);",
          "message": "MultiplicatorX4.withdraw (patch_5.sol#8-11) should be declared external\n"
        },
        {
          "name": "external-function",
          "vulnerability_from_line": 13,
          "vulnerability_to_line": 19,
          "vulnerability_code": "    function command(address adr, bytes data) public payable {\n        require(msg.sender == owner);\n        require(adr != address(0));\n\n        bool success = adr.call.value(msg.value)(data);\n        require(success);",
          "message": "MultiplicatorX4.command (patch_5.sol#13-19) should be declared external\n"
        },
        {
          "name": "external-function",
          "vulnerability_from_line": 21,
          "vulnerability_to_line": 26,
          "vulnerability_code": "    function multiplicate(address adr) public payable {\n        require(adr != address(0));\n        if (msg.value >= this.balance) {\n            adr.transfer(this.balance + msg.value);\n        }",
          "message": "MultiplicatorX4.multiplicate (patch_5.sol#21-26) should be declared external\n"
        },
        {
          "name": "solc-version",
          "vulnerability_from_line": 1,
          "vulnerability_to_line": null,
          "vulnerability_code": "pragma solidity ^0.4.18;",
          "message": "Detected issues with version pragma in patch_5.sol:\n\t- pragma solidity^0.4.18 (patch_5.sol#1): it allows old versions\n"
        },
        {
          "name": "low-level-calls",
          "vulnerability_from_line": 13,
          "vulnerability_to_line": 19,
          "vulnerability_code": "    function command(address adr, bytes data) public payable {\n        require(msg.sender == owner);\n        require(adr != address(0));\n\n        bool success = adr.call.value(msg.value)(data);\n        require(success);",
          "message": "Low level call in MultiplicatorX4.command (patch_5.sol#13-19):\n\t-success = adr.call.value(msg.value)(data) patch_5.sol#17\n"
        }
      ]
    },
    "mythril-0.23.15": {
      "successfull_analysis": true,
      "errors": [],
      "vulnerability_findings": [
        {
          "name": "External Call To User-Supplied Address (SWC 107)",
          "vulnerability_from_line": 17,
          "vulnerability_to_line": null,
          "vulnerability_code": "        bool success = adr.call.value(msg.value)(data);",
          "message": "A call to a user-supplied address is executed.\nAn external message call to an address specified by the caller is executed. Note that the callee account might contain arbitrary code and could re-enter any function within this contract. Reentering the contract in an intermediate state may lead to unexpected behaviour. Make sure that no state modifications are executed after this call and/or reentrancy guards are in place.\nClassification: SWC-107"
        }
      ]
    },
    "oyente": {
      "successfull_analysis": true,
      "errors": [],
      "vulnerability_findings": [
        {
          "name": "Callstack Depth Attack Vulnerability",
          "vulnerability_from_line": 17,
          "vulnerability_to_line": null,
          "vulnerability_code": "        bool success = adr.call.value(msg.value)(data);",
          "message": null
        }
      ]
    }
  }
}